### ISSUE TITLE
NamingStrategyInterface should allow enumName to be overridden

### ISSUE THREAD

**Issue by otavioleitao** [typeorm/typeorm#5572]
**Issue type:**

[ ] question
[ ] bug report
[x] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Context:**
TypeORM supports relationships between Entities and Enums. When defining this relationship with the `@Column` decorator, the `enumName` attribute can be set to define the Enum Type name in the database.

**Problem:**
The `NamingStrategyInterface` does not allow the naming strategy to be customised Enum Names. As it automatically generates types like `User_permissions_enum`.

Additionally, if you have two entities related to the same Enum, TypeORM creates two separate types on the database for that - what does not happen if you define the same enumName for both relationships.

**Suggested Solution:**
Allow the enum type naming to be defined in the `NamingStrategyInterface` by adding a new function to the interface.

**Comment by AlexMesser** [comment]
implemented in #4824. 

example: 

```ts
enum FooEnum {
    FOO,
    BAR,
}

class PostEntity {
    @PrimaryGeneratedColumn()
    id: number

    @Column({
        type: "enum",
        enum: FooEnum,
        enumName: "foo_enum",
    })
    foo: FooEnum
}
```

**Comment by mbrookson** [comment]
@AlexMesser While it's great the enum can be renamed, the change above doesn't actually resolve this issue. This specific issue is that a function should be provided on the `NamingStrategyInterface` to globally modify the naming convention used for all enums.
